### PR TITLE
Upgrade to Gradle 8

### LINKED ISSUE #131: Plugin application fails on Gradle 8.0
This fails since `AbstractArchiveTask#setClassifier` is dropped in Gradle 8.0.

https://github.com/unbroken-dome/gradle-testsets-plugin/blob/1c6fc3fa1d23af767c6c6fb276308adf89aaf2e9/src/main/kotlin/org/unbrokendome/gradle/plugins/testsets/TestSetsPlugin.kt#L133

### PR DESCRIPTION
**Pull request by neeme-praks-sympower**
* Upgrade to Gradle 8 (Gradle wrapper)
* Update `com.gradle.plugin-publish` plugin to `1.1.0` (previous version is not compatible with Gradle 8)
* Update publishing section of the build file to make it work with the new publishing plugin (at least it does not fail anymore)
* Fix Gradle plugin API compatibility issue in `TestSetsPlugin.kt`

Fixes #131

### CODE REVIEW

File: src/main/kotlin/org/unbrokendome/gradle/plugins/testsets/TestSetsPlugin.kt [hunk 1]
```diff
@@ -130,7 +130,7 @@ class TestSetsPlugin
             // Classifier is deprecated, but the new archiveClassifier was added only in Gradle 5.1
             // For compatibility we will still use the old one
             @Suppress("DEPRECATION")
```
Comment (catostrophe) [inline src/main/kotlin/org/unbrokendome/gradle/plugins/testsets/TestSetsPlugin.kt]:
```suggestion
```

**Comment by catostrophe** [comment]
BTW. If a Gradle plugin is built with Gradle 8, what minimal Gradle version it will be working with?

**Comment by neeme-praks-sympower** [comment]
> BTW. If a Gradle plugin is built with Gradle 8, what minimal Gradle version it will be working with?

I'm not a specialist but as far as I understand, as long as the plugin does not use any newer API, it will continue to work with Gradle 5.1 as before.
BTW, in hindsight I now realise that upgrade to Gradle 8 is not actually necessary to fix this API compatibility. I can remove that part of the PR, if you like.

**Comment by catostrophe** [comment]
I am not a maintainer and don't have an opinion regarding bumping the Gradle version. I just wanted to make sure that doing that won't make the plugin incompatible with Gradle 6+.

**Review by medgate-admin — APPROVED** [review]

**Comment by catostrophe** [comment]
@tkrullmann are you still maintaining this project?

**Review by lfarkas — APPROVED** [review]

**Comment by catostrophe** [comment]
This project seems abandoned. I would recommend migrating to [JVM Test Suite plugin]( https://docs.gradle.org/current/userguide/jvm_test_suite_plugin.html)

**Review by mderouet — APPROVED** [review]

**Review by mderouet — APPROVED** [review]

**Comment by piyushmor** [comment]
why is this still pending?

**Review by giovanni-nabil — APPROVED** [review]

**Comment by rd-janina-nemec** [comment]
Why is this not merged yet?

**Comment by lfarkas** [comment]
a release also needed...in order to be able to use it...

**Comment by pavelchuchma** [comment]
@tkrullmann any estimate of release with this fix, please?

**Pull request merged**